lib/ansible/modules/cloud/amazon/ec2_ami_facts.py

```diff
@@ -0,0 +1,237 @@
+    filters:
+      name: "ubuntu/images/ubuntu-zesty-17.04-*"
+'''
+
+RETURN = '''
+images:
+  description: a list of images
+  returned: always
+  type: complex
+  contains:
+    architecture:
+      description: The architecture of the image
+      returned: always
+      type: string
+      sample: x86_64
+    block_device_mappings:
+      description: Any block device mapping entries
+      returned: always
+      type: complex
+      contains:
+        device_name:
+          description: The device name exposed to the instance
+          returned: always
+          type: string
+          sample: /dev/sda1
+        ebs:
+          description: EBS volumes
+          returned: always
+          type: complex
+    creation_date:
+      description: The date and time the image was created
+      returned: always
+      type: string
+      sample: '2017-10-16T19:22:13.000Z'
+    description:
+      description: The description of the AMI
+      returned: always
+      type: string
+      sample: ''
+    ena_support:
+      description: whether enhanced networking with ENA is enabled
+      returned: always
+      type: bool
+      sample: true
+    hypervisor:
+      description: The hypervisor type of the image
+      returned: always
+      type: string
+      sample: xen
+    image_id:
+      description: The ID of the AMI
+      returned: always
+      type: string
+      sample: ami-5b466623
+    image_location:
+      description: The location of the AMI
+      returned: always
+      type: string
+      sample: 408466080000/Webapp
+    image_type:
+      description: The type of image
+      returned: always
+      type: string
+      sample: machine
+    name:
+      description: The name of the AMI that was provided during image creation
+      returned: always
+      type: string
+      sample: Webapp
+    owner_id:
+      description: The AWS account ID of the image owner
+      returned: always
+      type: string
+      sample: '408466080000'
+    public:
+      description: whether the image has public launch permissions
+      returned: always
+      type: bool
+      sample: true
+    root_device_name:
+      description: The device name of the root device
+      returned: always
+      type: string
+      sample: /dev/sda1
+    root_device_type:
+      description: The type of root device used by the AMI
+      returned: always
+      type: string
+      sample: ebs
+    sriov_net_support:
+      description: whether enhanced networking is enabled
+      returned: always
+      type: string
+      sample: simple
+    state:
+      description: The current state of the AMI
+      returned: always
+      type: string
+      sample: available
+    tags:
+      description: Any tags assigned to the image
+      returned: always
+      type: complex
+    virtualization_type:
+      description: The type of virtualization of the AMI
+      returned: always
+      type: string
+      sample: hvm
+'''
+
+import traceback
+
+try:
```

Comment:
No need to import boto3, and if you use AnsibleAWSModule, no need to import or test HAS_BOTO3.